content/doc/developer/testing/index.adoc

```diff
@@ -246,6 +246,48 @@ To recreate the same configuration from a test method, you can do the following:
 In order to test parts of your plugin, you may want certain files to exist in the build workspace, or that Jenkins is configured in a certain way.
 This section covers various ways to achieve this using the Jenkins Test Harness.
 
+=== Operating System Awareness
+
+Sometimes it ios necessary to adjust the testcase depending on the operating system.
+This can be the choice of Batch or Shell commands.
+The following example shows how to make use of `Functions.isWindows()` to create different code depending on the operating system:
+
+[source,java]
+----
+import hudson.Functions;
+import hudson.model.FreeStyleProject;
+import hudson.tasks.BatchFile;
+import hudson.tasks.Shell;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class MyPluginTest {
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    @Test
+    public void testOnAllSystems() throws Exception {
+        FreeStyleProject project = j.createFreeStyleProject();
+        if(Functions.isWindows()) {
+            project.getBuildersList().add(new BatchFile("echo ahoy > log.log"));
+        } else {
+            project.getBuildersList().add(new Shell("echo ahoy > log.log"));
+        }
+        j.buildAndAssertSuccess(project);
+    }
+}
+----
+
+If you just want to disable a test on Windows systems you can make use of `assumeFalse()` in combination with `Functions.isWindows()`:
```

Comment:
Proposed additional words to explain more about JUnit assumptions.
```suggestion
Sometimes a test should be skipped because it is only relevant for one type of operating system or because it needs specific conditions that depend on the test environment.
Tests can be skipped based on conditionals evaluated by link:https://www.baeldung.com/junit-conditional-assume[JUnit Assume].
Use  `assumeFalse()` to disable a test on Windows systems in combination with `Functions.isWindows()`:
```